### PR TITLE
Enable highlight colors for webdevicons

### PR DESCRIPTION
**Pull request by danhorner**
Thanks for this great and simple plugin.  This PR colorizes the tab icons by including the highlight groups  from [nvim-web-devicons](https://github.com/nvim-tree/nvim-web-devicons)

image

### CODE REVIEW

**Comment by Sangdol** [comment]
Thanks for the PR. The tabline in the screenshot looks pretty neat.

One concern is that the icon looks unnatural when I have a background color in my tabline since the icon itself doesn't have one. I searched a bit but it doesn't seem to be easy to set a background color to the icon, unless I'm missing something.

image

So, I think we should make the icon's colorization optional.

What do you think?

**Comment by Sangdol** [comment]
Sounds good. FYI, I'm going on vacation next week, so I might not answer for a while.

**Comment by danhorner** [comment]
Here's a small change that colors the tabs by creating an hlgroup for each icon.  

image

It was easiest to add this in the lua code. I experimented with caching the hlgroups instead of always generating them but there are only a dozen tabs open at a time and it just complicated things. Let me know if you prefer that approach.

File: lua/deviconutil.lua [hunk 1]
```diff
@@ -4,13 +4,35 @@
...
 local fn = vim.fn
 
-function M.get_icon(bufname)
+
+function M.get_icon(bufname, is_selected)
+  is_selected = is_selected ~= 0
   local loaded, webdev_icons = pcall(require, "nvim-web-devicons")
   if loaded then
-    return webdev_icons.get_icon(fn.fnamemodify(bufname, ":r"), fn.fnamemodify(bufname, ":e"), { default = true })
+    icon, hlgroup = webdev_icons.get_icon(fn.fnamemodify(bufname, ":r"), fn.fnamemodify(bufname, ":e"), { default = true })
```
Comment (Sangdol) [inline lua/deviconutil.lua:12]:
```suggestion
    local icon, hlgroup = webdev_icons.get_icon(fn.fnamemodify(bufname, ":r"), fn.fnamemodify(bufname, ":e"), { default = true })
```

**Review by Sangdol — APPROVED** [review]
Looks amazing. I couldn't think of this solution. Thanks for the contribution!

**Comment by Sangdol** [comment]
> I experimented with caching the hlgroups instead of always generating them but there are only a dozen tabs open at a time and it just complicated things. Let me know if you prefer that approach.

When I measured the time it took to draw the tabline, it was less than 0.001 seconds in my machine. Therefore, I think caching isn't worth it if it complicates things.

**Pull request merged**